### PR TITLE
Lesson 1 8

### PR DESCRIPTION
**Pull request by Victoria240**
Data Fetching, Re-Fetching, and Memoized Handlers

### CODE REVIEW

File: src/App.js
Comment (dargaCode) [inline src/App.js]: Now that you're fetching real data from Airtable, you no longer need to save nor load anything from `LocalStorage`! You can delete this effect entirely =D

File: src/App.js
Comment (dargaCode) [inline src/App.js]: Now that you're fetching real data from Airtable, you no longer need to save nor load anything from LocalStorage! You can delete this effect entirely =D

File: src/App.js
Comment (dargaCode) [inline src/App.js]: There's a small hidden bug here, which is that `data` is still an object with an array, and that array contains the actual todo that was created: 

Screen Shot 2023-08-24 at 12 09 06 AM

Comment (dargaCode) [reply src/App.js]: so to actually get the id and title confirmed by the server, you'll have to "unwrap" the layers of data around the todo: 

```suggestion
      const data = await response.json();

      const confirmedTodo = data.records[0];

      // Update the todoList state by adding the new todo
      setTodoList([...todoList, { id: confirmedTodo.id, title: confirmedTodo.fields.title }]);
```

Comment (dargaCode) [reply src/App.js]: I also want to point call attention to changing `newTodo.title` to `confirmedTodo.fields.title` - 

Because the backend is supposed to be the "source of truth" for all of our data, we should trust the title returned from Airtable over the title that we sent to Airtable, just in case they somehow become different. It won't make a difference in this case, but it's good to keep in mind for the future!

**Review by dargaCode — APPROVED** [review]
Great refactor, this is looking awesome now! 💪💪💪

I pointed out a couple small changes to make before you merge, but you're good to go after that!

File: src/App.js
Comment (Victoria240) [reply src/App.js:68]: Thank you for the feedback! I hope it's fine for me to leave it in here and merge this repo as it is, because I want to avoid any conflicts. I'll delete it in my next submission

**Pull request closed without merging**
